x-pack/test/api_integration/deployment_agnostic/services/role_scoped_supertest.ts

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  RoleCredentials,
+  SupertestWithoutAuthProviderType,
+  SamlAuthProviderType,
+} from '@kbn/ftr-common-functional-services';
+import { Test } from 'supertest';
+import { DeploymentAgnosticFtrProviderContext } from '../ftr_provider_context';
+
+export interface RequestHeadersOptions {
+  withInternalHeaders?: boolean;
+  withCommonHeaders?: boolean;
+  withCustomHeaders?: Record<string, string>;
+}
+
+export class SupertestWithRoleScope {
+  private roleAuthc: RoleCredentials | null;
+  private supertestWithoutAuth: SupertestWithoutAuthProviderType;
+  private samlAuth: SamlAuthProviderType;
+  private options: RequestHeadersOptions;
+  private destroyed = false;
```

Comment:
__question:__ Why do we need this additional field? It looks like the constructor of this class requires `RoleCredentials`, which cannot be `null`, and the only way to set `samlAuth` to `null` is by calling `destroy`. If that's the case, can't we just check if `samlAuth` is `null` or not?